src/upf/src/upf.i

```diff
@@ -106,4 +106,18 @@
                         x2,
                         y2);
   }
+
+  void set_power_switch_cell(char* name, char* cell)
+  {
+    odb::dbDatabase* db = getOpenRoad()->getDb();
+    upf::update_power_switch_cell(
+        getOpenRoad()->getLogger(), db->getChip()->getBlock(), name, cell); 
```

Comment:
a user most definitely would be to call this function before a Block or Chip has been created causing a segfault, same below. Probably needs to have some safeguards in the tcl

Reply:
I've added safeguards in the tcl. should be okay now